src/pybind/mgr/cephadm/module.py

```diff
@@ -1723,6 +1752,16 @@ def _create_daemon(self,
                 if spec.ports:
                     ports.extend(spec.ports)
 
+            if daemon_spec.daemon_type == 'haproxy':
+                haspec = cast(HA_RGWSpec, daemon_spec.spec)
+                if haspec.haproxy_container_image:
+                    image = haspec.haproxy_container_image
+
+            if daemon_spec.daemon_type == 'keepalived':
+                haspec = cast(HA_RGWSpec, daemon_spec.spec)
+                if haspec.keepalived_container_image:
+                    image = haspec.keepalived_container_image
```

Comment:
TBH, I'm not really a fan of having two daemons here. 

Would it be feasable to have both daemons in a single systemd unit? I mean, we have two containers for nfs and iscsi already.

Reply:
I think that to have separated systemd units for each daemon makes more easy to deal with any kind of unexpected problem, make things more granular, allows to have more flexibility in management , and avoid to affect one daemon with the problems in other daemon.

Reply:
we've discussed this yesterday in the standup. The idea is to:

* Separate services and daemons and make daemon_type != service_type a real thing. 
* introduce a new classification called kind: 
  * a spec of kind=service 
  * a spec of kind=daemon 
  * a spec of kind=host
  etc. 

This would make it possible to have compositions of daemons forming a single service.

Reply:
Seems sensible. I think that is a good idea but I continue thinking that systemd units should be different for different containers.

Reply:
I see a systemd unit as a k8s Pod. We have more than one container already for iscsi, osd and nfs.